mirdata/datasets/slakh.py

```diff
@@ -0,0 +1,332 @@
+            as calculated by the ITU-R BS.1770-4 spec
+        is_drum (bool): whether the "drum" flag is true for this MIDI track
+        midi_program_name (str): MIDI instrument program name
+        plugin_name (str): patch/plugin name that rendered the audio file
+        program_number (int): MIDI instrument program number
+
+    Cached Properties:
+        midi (PrettyMIDI): midi data used to generate the audio
+            Some unusual tracks have no midi - in this case this will be None
+        notes (NoteData): note representation of the midi data
+            Some unusual tracks have no audio - in this case this will be None
+
+    """
+
+    def __init__(self, track_id, data_home, dataset_name, index, metadata):
+
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name=dataset_name,
+            index=index,
+            metadata=metadata,
+        )
+
+        self.mtrack_id = self.track_id.split("-")[0]
+        self.audio_path = self.get_path("audio")
+        self.midi_path = self.get_path("midi")
+        self.metadata_path = self.get_path("metadata")
+
+        # split (train/validation/test/omitted) is part of the relative filepath in the index
+        self.data_split = None  # for baby_slakh, there are no data splits - set to None
+        if index["version"] == "2100-redux":
+            self.data_split = self._track_paths["audio"][0].split(os.sep)[1]
+            print(self._track_paths["audio"][0])
+            assert (
+                self.data_split in SPLITS
+            ), "{} not a valid split - should be one of {}.".format(
+                self.data_split, SPLITS
+            )
+
+    @core.cached_property
+    def _track_metadata(self) -> dict:
+        with open(self.metadata_path, "r") as fhandle:
+            metadata = yaml.safe_load(fhandle)
+        return metadata["stems"][self.track_id.split("-")[1]]
+
+    @property
+    def instrument(self) -> Optional[str]:
+        return self._track_metadata.get("inst_class")
+
+    @property
+    def integrated_loudness(self) -> Optional[float]:
+        return self._track_metadata.get("integrated_loudness")
+
+    @property
+    def is_drum(self) -> Optional[bool]:
+        return self._track_metadata.get("is_drum")
+
+    @property
+    def midi_program_name(self) -> Optional[str]:
+        return self._track_metadata.get("midi_program_name")
+
+    @property
+    def plugin_name(self) -> Optional[str]:
+        return self._track_metadata.get("plugin_name")
+
+    @property
+    def program_number(self) -> Optional[int]:
+        return self._track_metadata.get("program_num")
+
+    @core.cached_property
+    def midi(self) -> Optional[pretty_midi.PrettyMIDI]:
+        return io.load_midi(self.midi_path)
+
+    @core.cached_property
+    def notes(self) -> annotations.NoteData:
+        return io.load_notes_from_midi(self.midi_path, self.midi)
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
```

Comment:
for some tracks this is null and these tracks are added to the multitrack object. should we check in core.py if the audio is not null?

Reply:
We already do actually - it's handled in the io.coerce_from_bytes decorator!

Reply:
sorry! you're right!